packages/vite/src/node/plugins/css.ts

```diff
@@ -556,16 +543,13 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
           })
           chunk.viteMetadata.importedCss.add(this.getFileName(fileHandle))
         } else if (!config.build.ssr) {
-          // legacy build and inline css
-
-          // __VITE_ASSET__ and __VITE_PUBLIC_ASSET__ urls are processed by
-          // the vite:asset plugin, don't call resolveAssetUrlsInCss here
+          chunkCSS = resolveAssetUrlsInCss(chunkCSS, chunk.name, true)
           chunkCSS = await finalizeCss(chunkCSS, true, config)
-
           const style = `__vite_style__`
           const injectCode =
             `var ${style} = document.createElement('style');` +
-            `${style}.innerHTML = ${JSON.stringify(chunkCSS)};` +
+            `var relativeFromPage = (u) => new URL(u, self.location).href;` +
```

Comment:
We don't need this function here, as calling `renderAssetUrl` with hostType 'js' will do this for us.